src/plugins/topbar/topbar.jsx

```diff
@@ -19,14 +20,49 @@ export default class Topbar extends React.Component {
     this.setState({url: value})
   }
 
+  loadSpec = (url) => {
+    this.props.specActions.updateUrl(url)
+    this.props.specActions.download(url)
+  }
+
+  onUrlSelect =(e)=> {
+    let url = e.target.value || e.target.href
+    this.loadSpec(url)
+    e.preventDefault()
+  }
+
   downloadUrl = (e) => {
-    this.props.specActions.updateUrl(this.state.url)
-    this.props.specActions.download(this.state.url)
+    this.loadSpec(this.state.url)
     e.preventDefault()
   }
 
+  componentWillMount() {
```

Comment:
It's not clear to me what this method handles (with regards to the feature), or why it's needed. I removed it and my Topbar continued to select the first spec in `urls`, as I would expect.

It appears to read the browser query string for a name property, which is strange for two reasons:

- name isn't documented in the README as a config parameter
- components shouldn't read query strings directly, because it bypasses the YAML/constructor/qstring config inheritance system exposed via getConfigs.

Let me know if I'm reading this wrong.